src/adagrams.js

```diff
@@ -1,15 +1,97 @@
+// Implement this method for wave 1
+
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterBank = {'A':9, 'B':2, 'C':2, 'D':4, 'E':12, 'F':2, 'G':3, 'H':2, 'I':9, 'J':1,
+  'K':1, 'L':4, 'M':2, 'N':6, 'O':8, 'P': 2 , 'Q':1, 'R':6, 'S':4, 'T':6, 'U':4, 'V':2, 'W':2, 
+  'X':1, 'Y':2, 'Z':1}
+  
+
+  let handList = []
+  let tempHandobj = Object.assign({}, letterBank);
+
+  while (handList.length < 10) {
+    let randNum = Math.floor(Math.random() * 26);
+    let pickedLetter = Object.keys(tempHandobj)[randNum];
+    
+    if (tempHandobj[pickedLetter] !== 0){
+      handList.push(pickedLetter);
+      tempHandobj[pickedLetter] -= 1;
+    }
+  }
+  return handList;
+
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let word = input.toUpperCase()
+  let validInput = true;            
+  
+  for (const letter of word) {
+    if (!lettersInHand.includes(letter)) {
+      validInput = false;
+    }
+    lettersInHand.splice(0, 1, letter)
+  }
+  return validInput;
 };
 
-export const scoreWord = (word) => {
   // Implement this method for wave 3
+export const scoreWord = (word) => {
+
+  let upperWord = word.toUpperCase();
+  let score = 0;
+  
+  let list1 = ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'];
+  let list2 = ['D', 'G'];
+  let list3 = ['B', 'C', 'M', 'P'];
+  let list4 = ['F', 'H', 'V', 'W', 'Y'];
+  let list5 = ['K'];
+  let list8 = ['J', 'X'];
+  let list10 = ['Q', 'Z'];
```

Comment:
Why not make a JS object like this:

```javascript
const scores = {
  "AEIOULNRST": 1,
  "DG": 2,
...
}
```

Could you then do this without so many if stmts?  Alternatively you could use a dictionary were the keys are individual letters and the values being the corresponding scores.